src/Adapters/Storage/Mongo/MongoTransform.js

```diff
@@ -462,20 +462,33 @@ function untransformACL(mongoObject) {
 // cannot perform a transformation
 function CannotTransform() {}
 
+const transformInteriorAtom = atom => {
+  // TODO: check validity harder for the __type-defined types
+  if (typeof atom === 'object' && atom && !(atom instanceof Date) && atom.__type === 'Pointer') {
```

Comment:
I would extract the typeOf into a var ``let typeOfAtom = typeof atom`` then do all the comparisons against the var.

I remember reading something about throwing errors inside of functions cause the function to become very tricky for the JIT to improve, is that still the case with newer versions of node? Maybe it might be a idea to switch to returning ``CannotTransform`` Like some of the other functions do.


Maybe I'm crazy but this function is going to be called a lot right? for every value coming through the transformer? So to me this could be optimized so that number/string/plain objects don't have to go through all of the checks.

How about something like this?

```javascript
const transformInteriorAtom = atom => {
    if (atom == null) {
        return atom;
    }

    let typeofAtom = typeof atom;
    if (typeofAtom === 'function' || typeofAtom === 'symbol') {
       return CannotTransform;
    }

    if (typeofAtom !== 'object') {
        return atom;
    }

    if (atom.__type === 'Pointer') {
        return {
            __type: 'Pointer',
            className: atom.className,
            objectId: atom.objectId
        };
    }

    if (DateCoder.isValidJSON(atom)) {
        return DateCoder.JSONToDatabase(atom);
    }

    if (BytesCoder.isValidJSON(atom)) {
        return BytesCoder.JSONToDatabase(atom);
    }

    return atom;
}
```